### PR TITLE
Fix dependency issue in gsl (#24308)

### PR DESCRIPTION
**Pull request by glondu**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Thanks. Could you also apply same patch to all the previous affected versions? I believe all the dune/jbuilder versions are affected.

File: packages/gsl/gsl.1.24.3/opam [hunk 1]
```diff
@@ -37,3 +37,4 @@ url {
     "sha512=c6e2578618591d1bef428693b69026cdea0f1606cd25d9f02d637a90256a5685eee70ecd0259d2595a1cd7b292c34e3c913c007e2706aa125af045d37e55d9c4"
   ]
 }
+patches: ["fixdeps.patch"]
```
Comment (kit-ty-kate) [inline packages/gsl/gsl.1.24.3/opam:40]: The extra-files field is missing
```suggestion
patches: ["fixdeps.patch"]
extra-files: ["fixdeps.patch" "md5=???"]
```

**Comment by glondu** [comment]
> Thanks. Could you also apply same patch to all the previous affected versions? I believe all the versions with dune/jbuilder are affected.

...which means testing/adapting the patch to all those versions? It is not part of my QA routine... so I won't bother/care.

**Comment by kit-ty-kate** [comment]
Ok no worries. Thanks!

**Pull request merged**